direct_chat.py

```
@@ -461,6 +472,12 @@ async def _handle_regular_chat(
             "response": assistant_message,
             "state": DirectChatState.ASSISTANT_REPLY
         })
+    except asyncio.TimeoutError:
+        log.error("Direct chat timed out for user %s", user.email)
+        raise HTTPException(
+            status_code=504,
+            detail="The AI service took too long to respond. Please try again or select a different model.",
+        )
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)